source/templates/writing-helpers.md

```diff
@@ -218,24 +218,44 @@ And two named arguments:
 * `format="YYY MM DD h:mm"`
 * `locale="en"`
 
-### Stateful Helpers
+### Service Helpers
 
 By default, helpers are _stateless_. They are passed inputs (parameters
 and a hash), they perform an operation on those inputs, and return a
 single output. They have no side-effects and don't save any information
 that is used on subsequent runs of the function.
 
-In some situations, however, you may need to write a helper that
-interacts with the rest of your application. You can create stateful
-helpers that have access to services in your application, and can
-optionally save state as well.
+In some situations, however, you may need to write a helper that interacts with
+the rest of your application. You can create stateful service helpers that have
+access to services in your application, and can optionally save state as well
+(this is usually unnecessary and error-prone).
 
-To create a stateful helper, rather than returning a simple function,
-you should return a subclass of `Ember.Helper`. Helper classes must
-contain a `compute` method that behaves the same as the function passed
-to `Ember.Helper.helper`.
+To create a service helper, rather than exporting a simple function, you
```

Comment:
issue raised in https://github.com/emberjs/guides/pull/805/files#r40445352

(just marking them to make sure they get addressed)